src/admin-views/aggregator/settings.php

```diff
@@ -142,6 +142,51 @@
 			'parent_option' => Tribe__Events__Main::OPTIONNAME,
 			'options' => $yes_no_options,
 		),
+		'tribe_aggregator_default_import_limit_type' => array(
+			'type' => 'dropdown',
+			'label' => esc_html__( 'Import limit type', 'the-events-calendar' ),
+			'tooltip' => esc_html__( 'Limit the number of imported events by number, date range or not at all; on slower websites this may impact the success of imports. Selecting a shorter time period or a smaller number  of events may improve results.', 'the-events-calendar' ) . ' ' . sprintf( '<a href="%1$s" target="_blank">%2$s</a>', esc_attr( 'https://theeventscalendar.com/knowledgebase/other-url-import-errors-in-event-aggregator' ), esc_html( 'Learn more.' ) ),
+			'size' => 'medium',
+			'validation_type' => 'options',
+			'default' => 'by_range',
+			'can_be_empty' => false,
+			'parent_option' => Tribe__Events__Main::OPTIONNAME,
+			'options' => tribe( 'events-aggregator.settings' )->get_import_limit_type_options(),
+		),
+		'tribe_aggregator_default_import_limit_range' => array(
+			'type' => 'dropdown',
+			'label' => esc_html__( 'Import date range limit', 'the-events-calendar' ),
+			// @todo set the KB link to the correct link
+			'tooltip' => esc_html__( 'When importing from an event source import events this far into the future; on slower websites this may impact the success of imports. Selecting a shorter time period may improve results.', 'the-events-calendar' ) . ' ' . sprintf( '<a href="%1$s" target="_blank">%2$s</a>', esc_attr( 'https://theeventscalendar.com/knowledgebase/other-url-import-errors-in-event-aggregator' ), esc_html( 'Learn more.' ) ),
+			'size' => 'medium',
+			'validation_type' => 'options',
+			'default' => tribe( 'events-aggregator.settings' )->get_import_range_default( true ),
+			'can_be_empty' => true,
+			'parent_option' => Tribe__Events__Main::OPTIONNAME,
+			'options' => tribe( 'events-aggregator.settings' )->get_import_range_options( true ),
+			'class' => 'tribe-dependent',
+			'fieldset_attributes' => array(
+				'data-depends'   => '#tribe_aggregator_default_import_limit_type-select',
+				'data-condition' => 'range',
+			),
+		),
+		'tribe_aggregator_default_import_limit_number' => array(
+			'type' => 'dropdown',
+			'label' => esc_html__( 'Import numeric limit', 'the-events-calendar' ),
+			// @todo set the KB link to the correct link
+			'tooltip' => esc_html__( 'When importing from an event source import this many events; on slower websites this may impact the success of imports. Setting this to a smaller number may improve results.', 'the-events-calendar' ) . ' ' . sprintf( '<a href="%1$s" target="_blank">%2$s</a>', esc_attr( 'https://theeventscalendar.com/knowledgebase/other-url-import-errors-in-event-aggregator' ), esc_html( 'Learn more.' ) ),
```

Comment:
How about:

```
When importing from an event source, this is the maximum number of events that will be imported; on slower websites this may impact the success of imports. Setting this to a smaller number may improve results.
```

Reply:
See above for my questions on the KB article.